examples/b2c/checkout/integrations/classes/B2CDeliverySample.cls

```diff
@@ -241,19 +280,35 @@ global class B2CDeliverySample implements sfdc_checkout.CartShippingCharges {
         // ReferenceNumber: Reference Number from External Service
         // IsActive: If this Option is Active
 
+        // Below fields are available only for api version >= 61
+
+        // TransitTimeMin: Minimum Transit Time
+        // TransitTimeMax: Maximum Transit Time
+        // TransitTimeUnit: Time Unit for Transit Time (Valid Values are "Hours", "Days", "Weeks")
+        // ProcessTime: Process Time
+        // ProcessTimeUnit: Time Unit for Process Time (Valid Values are "Hours", "Days", "Weeks")
+
         Id productId = getDefaultShippingChargeProduct2Id();
-        CartDeliveryGroupMethod cartDeliveryGroupMethod = new CartDeliveryGroupMethod(
-            CartDeliveryGroupId = cartDeliveryGroupId,
-            ExternalProvider = shippingOption.getProvider(),
-            Name = shippingOption.getName(),
-            ShippingFee = shippingOption.getRate(),
-            WebCartId = webCartId,
-            Carrier = shippingOption.getCarrier(),
-            ClassOfService = shippingOption.getClassOfService(),
-            ProductId = productId,
-            ReferenceNumber = shippingOption.getReferenceNumber(),
-            IsActive = shippingOption.isActive()
-        );
+
+        CartDeliveryGroupMethod cartDeliveryGroupMethod = new CartDeliveryGroupMethod();
+        cartDeliveryGroupMethod.put('CartDeliveryGroupId', cartDeliveryGroupId);
+        cartDeliveryGroupMethod.put('ExternalProvider', shippingOption.getProvider());
+        cartDeliveryGroupMethod.put('Name', shippingOption.getName());
+        cartDeliveryGroupMethod.put('ShippingFee', shippingOption.getRate());
+        cartDeliveryGroupMethod.put('WebCartId', webCartId);
+        cartDeliveryGroupMethod.put('Carrier', shippingOption.getCarrier());
+        cartDeliveryGroupMethod.put('ClassOfService', shippingOption.getClassOfService());
+        cartDeliveryGroupMethod.put('ProductId', productId);
+        cartDeliveryGroupMethod.put('ReferenceNumber', shippingOption.getReferenceNumber());
+        cartDeliveryGroupMethod.put('IsActive', shippingOption.isActive());
+
+        if(apexClass.ApiVersion >= 61) {
```

Comment:
Is this the typical way to do this? What if the apex is newer version but the source code is older?

Reply:
If the source code is older, there will be no option to select the API version for the Apex classes beyond the limit.